app/uk/gov/hmrc/gform/gform/EmailAuthController.scala

```diff
@@ -344,6 +350,21 @@ class EmailAuthController(
               MarkDownUtil.markDownParser(emailConfirmation)
             )
           ).pure[Future]
+        case Composite(configs) if configs.exists(_.isEmailConfirmation) =>
```

Comment:
There is a bit of duplication here. What about introducing `HasEmailConfirmation` extractor so instead of these two cases

```
case EmailAuthConfig(_, _, _, Some(emailConfirmation))
case Composite(configs) if configs.exists(_.isEmailConfirmation)
```

We would need to do only:

```
case HasEmailConfirmation(emailConfirmation)
```

?